java/sendMessage/Index.java

```diff
@@ -0,0 +1,174 @@
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import io.github.redouane59.twitter.TwitterClient;
+import io.github.redouane59.twitter.signature.TwitterCredentials;
+
+private static class MessageSendingFailedException extends RuntimeException {
+    public MessageSendingFailedException(String message) {
+        super(message);
+    }
+
+    public MessageSendingFailedException(String message, Throwable cause) {
+        super(message, cause);
+    }
+}
+
+public RuntimeResponse main(RuntimeRequest req, RuntimeResponse res) {
+    Map<String, BiConsumer<Map<String, Object>, Map<String, String>>> messageDispatcher = Map.of(
+            "Discord", this::sendDiscordMessage,
+            "Email", this::sendEmailMessage,
+            "SMS", this::sendSMSMessage,
+            "Twitter", this::sendTwitterMessage
+    );
+
+    String payloadString = req.getPayload();
+    Gson gson = new Gson();
+    Map payload = gson.fromJson(payloadString, Map.class);
+
+    try {
+        Optional.ofNullable(messageDispatcher.get((String) payload.get("type")))
+                .orElseThrow(() -> new IllegalArgumentException(String.format("No such type: %s", payload.get("type"))))
+                .accept(payload, req.getVariables());
+        return res.json(Map.of("success", true));
+    } catch (IllegalArgumentException exception) {
+        return res.json(Map.of(
+                "success", false,
+                "message", exception.getMessage()
+        ));
+    }
```

Comment:
This doesn't catch an exception if the API call fails due to a bad API key. Instead, the output is:

```json
{
  "stdout": "",
  "stderr": "io.openruntimes.java.Wrapper$MessageSendingFailedException: Request failed: Forbidden\n\tat io.openruntimes.java.Wrapper.sendEmailMessage(Wrapper.java:109)\n\tat io.openruntimes.java.Wrapper.main(Wrapper.java:45)\n\tat io.openruntimes.java.Server.execute(Server.java:43)\n\tat org.rapidoid.http.handler.optimized.DelegatingParamsAwareReqRespHandler.handleReq(DelegatingParamsAwareReqRespHandler.java:49)\n\tat org.rapidoid.http.handler.AbstractHttpHandlerDecorator.handleReqAndPostProcess(AbstractHttpHandlerDecorator.java:48)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator$3.invokeNext(HttpManagedHandlerDecorator.java:161)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator$3.invoke(HttpManagedHandlerDecorator.java:121)\n\tat org.rapidoid.http.handler.HttpAuthWrapper.wrap(HttpAuthWrapper.java:70)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator.wrap(HttpManagedHandlerDecorator.java:185)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator.handleWithWrappers(HttpManagedHandlerDecorator.java:100)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator.access$200(HttpManagedHandlerDecorator.java:39)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator$2.run(HttpManagedHandlerDecorator.java:83)\n\tat org.rapidoid.job.PredefinedContextJobWrapper.run(PredefinedContextJobWrapper.java:56)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"
}
```